pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/HashJoinOperator.java

```diff
@@ -207,13 +207,17 @@ protected TransferableBlock getNextBlock()
       buildBroadcastHashTable();
     }
     if (_upstreamErrorBlock != null) {
+      LOGGER.trace("Returning upstream error block for join operator");
```

Comment:
Yes, all the changes are just new logs I've added while debuging some blocking issues. These logs are disabled by default and should be easily removed by either the jit given they are not allocating. I recommend to keep them in case we need to debug something in the future.